spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/writer/BoundStatementBuilder.scala

```diff
@@ -38,11 +39,18 @@ private[connector] class BoundStatementBuilder[T](
       val converter = converters(i)
       val columnName = columnNames(i)
       val columnValue = converter.convert(buffer(i))
-      boundStatement.set(columnName, columnValue, CodecRegistryUtil.codecFor(columnTypes(i), columnValue))
-      val serializedValue = boundStatement.getBytesUnsafe(i)
-
-      if (serializedValue != null)
-        bytesCount += serializedValue.remaining()
+      //C* 2.2 (PV4) and Greater Allows us to leave fields Unset
+      if (protocolVersion.toInt < ProtocolVersion.V4.toInt || columnValue != Unset) {
```

Comment:
I think we can make a function reference and assign one implementation or another once `BoundStatementBuilder` is created, and do not repeat here.